logprep/processor/domain_label_extractor/factory.py

```diff
@@ -2,11 +2,14 @@
 
 from logprep.processor.base.factory import BaseFactory
 from logprep.processor.domain_label_extractor.processor import DomainLabelExtractor
+from logprep.processor.processor_factory_error import InvalidConfigurationError
 
 
 class DomainLabelExtractorFactory(BaseFactory):
     """Factory used to instantiate DomainLabelExtractor processors."""
 
+    mandatory_fields = ["type", "generic_rules", "specific_rules", "tree_config"]
```

Comment:
what is with the `tld_lists` ... are they optional?

Reply:
They are. [https://logprep.readthedocs.io/en/latest/user_manual/configuration/processor.html#domain-label-extractor](Logprep Docs)